twtxt/helper.py

```diff
@@ -85,3 +87,28 @@ def sort_and_truncate_tweets(tweets, direction, limit):
             return sorted(tweets)
     else:
         return []
+
+
+@contextlib.contextmanager
+def atomic_write(path, mode):
+    """
+        Simple contextmanager to safely write to a
+        file.
+
+        :param str filename: the path to write to
+        :param str mode: the mode to open the file
+    """
+    # generate random filepath in same location as given filename
+    tmpfilepath = os.path.join(os.path.dirname(path), ".{}".format(uuid.uuid4()))
```

Comment:
Is there a way to leverage python's built-in tempfile support for this? https://docs.python.org/2/library/tempfile.html

It's possible there may not be. The temporary file has to be on the same filesystem as the 'real' file, and I don't know if tempfile has bits that can ensure that or not. Just a suggestion!